### DISCUSSION TITLE
Import CSV from MyBookshelf

### DISCUSSION THREAD

**Discussion by geekgens** [mateusz-bak/openreads#721] [Ideas]
**Is your feature request related to a problem? Please describe.**

因为[Mybookshelf](https://github.com/smartjinyu/MyBookshelf)没更新，而且我在[issue](https://github.com/smartjinyu/MyBookshelf/issues/17)上看到有人推荐这个，我认为open library的图书库不全，很多书在里面没找到。比如中国的书，希望能添加其他书库来源。比如豆瓣读书api(api接口可以参考这个用https://github.com/caspartse/eBooksAssistant?tab=readme-ov-file)
Because [Mybookshelf](https://github.com/smartjinyu/MyBookshelf) has not been updated, and I saw someone recommending this on [issue](https://github.com/smartjinyu/MyBookshelf/issues/17), I think the open library library is incomplete and many books cannot be found in it. For example, for Chinese books, I hope to add other library sources. For example, Douban Reading API (for the API interface, please refer to this https://github.com/caspartse/eBooksAssistant?tab=readme-ov-file)

**Describe the solution you'd like**

[Mybookshelf](https://github.com/smartjinyu/MyBookshelf)导出csv在openreads导入没效果，可以兼容下，应该是图书在open library上搜不到的问题
[Mybookshelf](https://github.com/smartjinyu/MyBookshelf) Exporting csv and importing it into openreads has no effect. It can be compatible. It should be a problem that the book cannot be found on open library.

**Describe alternatives you've considered**

可以添加中国方面的图书库，比如豆瓣读书的api，但是官方是付费的，不过有个开源的可以支持每天1000次的免费查询.(https://github.com/caspartse/eBooksAssistant?tab=readme-ov-file)，不知道作者能不能找到更好的图书数据查询
You can add Chinese book libraries, such as Douban Reading API, but the official one is paid, but there is an open source one that can support 1,000 free queries per day. (https://github.com/caspartse/eBooksAssistant?tab=readme -ov-file), I wonder if the author can find a better book data query

我提供下我备份的[Mybookshelf](https://github.com/smartjinyu/MyBookshelf)的csv格式文件
I provide the csv format file of [Mybookshelf](https://github.com/smartjinyu/MyBookshelf) that I backed up.
[Bookshelf_CSV.csv](https://github.com/mateusz-bak/openreads/files/14603200/Bookshelf_CSV.csv)

**Additional context**

**Comment by mateusz-bak** [comment]
There no option to import Mybookshelf csv into openreads, so I gues this is asing for this feature?

If yes can you provide any documentation about the CSV that is exported from mybookshelf?

**Comment by geekgens** [comment]
Yes,Many books (Chinese publications) cannot be found on Open Library. It feels like we need to add other search sources besides Open Library.

The format of Mybookshelf CSV is as follows:
"Number","Title","Author","Translator","Publisher","Publication Date","ISBN","Reading Status","Bookshelf","Tags","Notes","Website"

Also, I'll provide some ISBNs of books that cannot be found on Open Library for your reference: 9787121133961, 9787302029014.

I've also provided the CSV file that I exported.
[Bookshelf_CSV.csv](https://github.com/mateusz-bak/openreads/files/14915489/Bookshelf_CSV.csv)
